composer/metrics/nlp.py

```diff
@@ -604,51 +605,62 @@ def update(self, batch: Dict[str, Any], outputs: List[str], labels: List[str]):
             labels (List[str]): A list of the correct code generations, for compatibility with existing HF generate
             functionalities. This is not used.
         """
+        if not self._initialized:
+            # we need to defer the accumulator initialization because it depends on dataset size
+            dataset_size = batch['dataset_size']
+            device = batch['input_ids'].device
+            self.add_state('correct', default=torch.zeros(dataset_size, device=device), dist_reduce_fx='sum')
```

Comment:
how come the tensors in the state are now on device? were they just on cpu before? is there any reason from them to be on gpu?

Reply:
good point, got carried away. Addressed